src/Notepads/Utilities/FileSystemUtility.cs

```diff
@@ -135,6 +136,43 @@ public static async Task<StorageFile> OpenFileFromCommandLine(string dir, string
             return await GetFile(path);
         }
 
+        private static string ReplaceEnvironmentVariables(string args)
+        {
+            if (args.Contains("%homepath%", StringComparison.OrdinalIgnoreCase))
+            {
+                args = args.Replace("%homepath%",
+                    Environment.GetFolderPath(Environment.SpecialFolder.UserProfile),
+                    StringComparison.OrdinalIgnoreCase);
+            }
+
+            if (args.Contains("%localappdata%", StringComparison.OrdinalIgnoreCase))
+            {
+                args = args.Replace("%localappdata%",
+                    UserDataPaths.GetDefault().LocalAppData,
+                    StringComparison.OrdinalIgnoreCase);
+            }
+
+            if (args.Contains("%temp%", StringComparison.OrdinalIgnoreCase))
+            {
+                args = args.Replace("%temp%",
+                    (string)Microsoft.Win32.Registry.GetValue(@"HKEY_CURRENT_USER\Environment",
```

Comment:
Do we need extra permission to access registry?

Reply:
No, reading registry doesn't require any permission.